File: 57-insert-interval/insert-interval.cpp
class Solution {
public:
    
    vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) {

        vector<vector<int>> res;
        bool isInserted = false;

        for(int i = 0; i < intervals.size(); i++){
            vector<int>interval = intervals[i];
            if(interval[1] < newInterval[0]){
                res.push_back(interval);
            }else if(interval[0] > newInterval[1]){
                if(!isInserted){
                    res.push_back(newInterval);
                    isInserted = true;
                } 
                res.push_back(interval);
            }else{
                newInterval[0] = min(newInterval[0], interval[0]);
                newInterval[1] = max(newInterval[1], interval[1]);
            }
        }

        if(!isInserted) res.push_back(newInterval);

        return res;
    }
};

File: 57-insert-interval/insert-interval.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        res = []
        isInserted = False

        for interval in intervals:
            if interval[1] < newInterval[0]:
                res.append(interval)
            elif interval[0] > newInterval[1]:
                if not isInserted:
                    res.append(newInterval)
                    isInserted = True
                res.append(interval)
            else:
                newInterval[0] = min(interval[0], newInterval[0])
                newInterval[1] = max(interval[1], newInterval[1])
        
        if not isInserted:
            res.append(newInterval)

        return res